app/src/main/java/com/e/salesapp/ProductDetailsActivity.java

```diff
@@ -5,27 +5,30 @@
 
 import androidx.appcompat.app.AppCompatActivity;
 
+import com.squareup.picasso.Picasso;
 import com.synnapps.carouselview.CarouselView;
 import com.synnapps.carouselview.ImageListener;
 
 public class ProductDetailsActivity extends AppCompatActivity {
 
-    CarouselView carouselView;
-    int[] sampleImages = {R.drawable.fruits, R.drawable.fruits, R.drawable.fruits};
+    CarouselView mCarouselView;
+    String[] sampleImages = {"https://www.worldatlas.com/r/w728-h425-c728x425/upload/46/cb/e1/shutterstock-252338818.jpg",
+                            "https://previews.123rf.com/images/aiselin/aiselin1809/aiselin180900081/107486908-lots-of-fruits-strawberries-blueberries-mango-orange-grapefruit-banana-apple-grapes-kiwis-on-the-whi.jpg",
+                            "https://previews.123rf.com/images/cooperr007/cooperr0071711/cooperr007171100268/89193608-image-of-a-lot-of-fruits-and-vegetables-closeup.jpg"};
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_product_details);
-        carouselView = (CarouselView) findViewById(R.id.carouselView);
-        carouselView.setPageCount(sampleImages.length);
-        carouselView.setImageListener(imageListener);
+        mCarouselView = (CarouselView) findViewById(R.id.carouselView);
+        mCarouselView.setPageCount(sampleImages.length);
+        mCarouselView.setImageListener(imageListener);
```

Comment:
I don't like this implementation, my suggestion is change to ViewPager and inside it implement the ViewHolder pattern it will help you to handle better the device memory and application performance